docs/setup.asciidoc

```diff
@@ -0,0 +1,66 @@
+[[setup]]
+== Set up the Agent
+
+[float]
+=== Gradle setup
+
+First, add the Elastic APM agent plugin to your application's `build.gradle` file as shown below:
+
+[source,groovy]
+----
+// Android app's build.gradle file
+plugins {
+    id "com.android.application"
+    id "co.elastic.apm.android" version "[latest_version]" <1>
+}
+----
+<1> The Elastic plugin declaration must be added below the Android app plugin declaration (`com.android.application`)
+and below the Kotlin plugin declaration (if used).
+
+After adding the agent plugin, configure it.
+A minimal configuration sets the Elastic APM Server endpoint as shown below:
+
+[source,groovy]
+----
+// Android app's build.gradle file
+plugins {
+    //...
+    id "co.elastic.apm.android" version "[latest_version]"
+}
+
+elasticApm {
+    // Minimal configuration
+    serverUrl = "https://your.elastic.server"
+
+    // Optional
+    serviceName = "your app name" <1>
+    serviceVersion = "0.0.0" <2>
+    secretToken = "your server auth token" <3>
+}
+----
+<1> Defaults to your `android.defaultConfig.applicationId` value.
+<2> Defaults to your `android.defaultConfig.versionName` value.
+<3> Defaults to null.
+
+[float]
+=== Application setup
+
+After syncing your project with the Gradle changes above, the Elastic APM agent needs to be initialized within your https://developer.android.com/reference/android/app/Application[Application class].
+This example shows the simplest way to configure the agent:
+
+[source,java]
+----
+// Your Application class
+
+class MyApp extends android.app.Application {
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+        ElasticApmAgent.initialize(this); <1>
+    }
+}
+----
+<1> Initialize the Elastic APM agent once.
+
```

Comment:
So step one is to add the plugin declaration to Gradle. Step two is initialize the agent in the Application class. What's step three? Does it just work™️? Is an application restart required?

Reply:
That's a great point, I've been doing this for so long that I didn't think of how the overall init process is perceived the first time 😅
So, if you want to go with the default behaviour (which is to automatically instrument http requests and UI rendering as shown on the "supported technologies"), then yes, this is all you need to do because, step 3 might be to just compile (since it's an app, I think there's no "restart" concept, but instead a recompile or just compile process), which is something users will do anyway, but now that you mention it, I think it's better to clarify what they'll get out of those 2 steps and link them to the config in case they'd like to do some optional steps